search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/DefaultLlmImpl.java

```diff
@@ -136,6 +136,19 @@ protected Map<String, String> getInputParameters(ChatCompletionInput chatComplet
                             chatCompletionInput.getContexts()
                         )
                 );
+        } else if (chatCompletionInput.getModelProvider() == ModelProvider.BEDROCK_CONVERSE) {
```

Comment:
Instead of keep adding new "else if" blocks here, why not just define an annotation for different methods and use reflection to process different models servers at runtime? In that way I think we could reduce the code complexity here.

Reply:
Given the time constraint, I would do the refactoring at a later time.  I have been thinking about a good way to avoid this style of handling different cases of LLM vendors and APIs, but I was waiting for some general patterns to emerge which I think is this Message API, but again there are still small differences between LLM providers.